docs/authentication.md

```diff
@@ -0,0 +1,211 @@
+# Authentication Functions
+
+<!-- Glee relies on functions to execute your business logic. Functions are files that export a default async Node.js function: -->
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory, the name of the authentication file should be the name of the targeted server that the authentication logic should work for.
+
+## Supported Authentication Values in AsyncAPI.yaml file
+
+AsyncAPI currently supports a variety of authentication formats as specified in the documentation, however Glee supports the following authentication schemas.
```

Comment:
add a link to the said documentation. (I mean the spec docs that specifies the types of authentications that asyncapi supports)